new/missing-method-missing.md

```diff
@@ -0,0 +1,213 @@
+`provides` directives to direct Chef to the correct resource class, the
+namespaces `Chef::Resource` is explicitly searched for
+`Chef::Resource::MyResource`.
+
+This means that if you have seen examples of classes in Chef::Resource, and want
+to do the obvious thing and just move to your own namespace, it doesn't
+immediately work and it is far from apparent why. The result? A massive traffic
+jam of non-core resources in the `Chef::Resource` namespace and puzzling error
+messages for cookbook conflicts such as "superclass mismatch for class MyResource."
+
+Chef will now automatically call `provides` on all new, named resource classes,
+using `ResourceClass.dsl_name` (a transform of the class name, i.e.
+MyPackage::MyResource -> my_resource) as the provided class.
+
+```ruby
+class MyPackage::MyResource < Chef::Resource
+end
+# now you can type 'my_resource' in a recipe.
+```
+
+#### Explicit vs. Implicit `provides`
+
+If you specify `provides :something_else` in a class, that will override the
+auto-class-creation behavior.
+
+```ruby
+class MyPackage::MyResource < Chef::Resource
+  provides :shazam
+end
+# Now `shazam` will work fine in a recipe, but `my_resource` will not.
+```
+
+If you want to still provide the original name, you must be explicit:
+
+```ruby
+class MyPackage::MyResource < Chef::Resource
+  provides :my_resource
+  provides :shazam
+end
+# Now both `shazam` and `my_resource` work in a recipe
+```
+
+#### `does_not_provide`
+
+If you want to (for example) create a base class that does not have an actual
+DSL method, you can specify `does_not_provide` or `does_not_provide :my_name` in
+the class:
+
+```ruby
+class MyPackage::MyResourceBase < Chef::Resource
+  does_not_provide
+end
+# my_resource_base will not exist in recipe DSL
+```
+
+### Move LWRPs out of the Chef::Resource namespace
+
+With Chef::Resource no longer a special place for class lookup, LWRPs no longer
+need to be given a namespace.  We will now make resources/x.rb and providers/x.rb
+an anonymous class, with string methods to make it easy to see where they come
+from.  This avoids a number of potential errors and warnings that can happen in
+the case of conflicts, and gives us control over how we handle conflicts.
+
+Users who wish to set the class of the resource can assign it directly in the
+LWRP with `Namespace::MyResource = self`.
+
+## Deprecations and Errors
+
+Removing `method_missing` means several methods of creating resource names cannot
+be caught.  These are the cases.  We will issue deprecation warnings for each one
+and remove them (and method_missing) come Chef 13.
+
+### Chef::Resource::MyResource = <resource>
+
+When you assign an anonymous class or a resource from another namespace into
+`Chef::Resource`, your new resource will currently be accessible via the DSL.
+Since there is no Ruby method that lets us catch the assignment of a constant,
+there is no way for us to assign an explicit DSL method to handle the resource.
```

Comment:
To be clear, this section of the RFC could *not* go away.  It still has to work in Chef 12, and be marked deprecated.  It would just be a different reason (classes don't get implicit values).

*Also* to be clear, the logic implementing this would go away in Chef 13 in *both* cases.  There is no net gain for removing the implicit gain, as far as this goes.